grammar.js

```diff
@@ -26,33 +43,59 @@ module.exports = grammar({
   name: "git_commit",
 
   extras: ($) => [WHITE_SPACE],
-
   rules: {
     source: ($) =>
       seq(
-        optional(choice($.comment, $.subject)),
+        optional(choice($.subject, $.comment)),
         repeat($._body_line),
+        repeat($._trailer),
+        optional(repeat1($._newline)),
         optional(
           seq(alias(SCISSORS, $.scissors), optional(alias($._rest, $.message)))
         )
       ),
-
-    _body_line: ($) => seq(/[\r\n]+/, optional(choice($.message, $.comment))),
-
+    _newline: ($) => /\r?\n/,
+    /**
+     * The subject of the commit message: the first line.
+     */
     subject: ($) =>
-      seq(token(prec(PREC.SUBJECT, /[^#\r\n]/)), repeat(ANYTHING)),
+      seq(
+        prec(
+          LEXICAL_PRECEDENCE.SUBJECT_FIRST_CHAR,
+          token.immediate(/[^#\r\n]/)
+        ),
```

Comment:
```suggestion
        token.immediate(
          prec(LEXICAL_PRECEDENCE.SUBJECT_FIRST_CHAR, /[^#\r\n]/)
        ),
```

If I understand correctly, `prec` needs to be used within a `token` rule for this to be lexical precedence

Reply:
That raises an interesting point: I'm able to remove the precedence entirely and still pass all the tests.

Reply:
Hmm yeah I don't quite remember why that prec is there to begin with. Removing it should be fine if it doesn't break any tests 👍

Reply:
done in [9c7e616](https://github.com/the-mikedavis/tree-sitter-git-commit/pull/8/commits/9c7e616df9bc1ba50aa8a58ee68866a2a041bc55)